batnode.js

```diff
@@ -250,25 +251,59 @@ class BatNode {
       this.getHostNode(currentCopy, afterHostNodeIsFound)
     }
   }
+  combineShardsAfterWaitTime(waitTime, fileName, distinctShards) {
+    return new Promise(resolve => {
```

Comment:
I think Node sends strong warnings about not handling promise rejections (i.e not using the `reject` param) saying that the process will crash in future versions of node if that happens. Can we try and handle it here?

Reply:
You mean adding `if (error) reject(error);` in `Promise` to handle it?

Reply:
Often you can do something like that, yeah. I don't think your `combineShardsAfterWaitTime` takes an error parameter, but maybe another method does. Another option might be to think about what could happen in the function body logic that could cause an error, like trying to read a file that doesn't exist. I'm not an expert, so could be worth reading about promise rejection stuff, too.

Reply:
Add reject handling in Promise: bd603b1346fbb94cdcb98e008ac179f3617a6d18

Reply:
This line [here](https://github.com/layr-team/Layr/blob/bd603b1346fbb94cdcb98e008ac179f3617a6d18/kad-bat-plugin/batnode.js#L269) depends on an error being passed in if present, but it looks like `asyncCallAssembleShards` is called [here](https://github.com/layr-team/Layr/blob/bd603b1346fbb94cdcb98e008ac179f3617a6d18/kad-bat-plugin/batnode.js#L303) without an error param. [Here](https://github.com/layr-team/Layr/blob/bd603b1346fbb94cdcb98e008ac179f3617a6d18/kad-bat-plugin/batnode.js#L266) an error param is passed in. I guess, how would `combineShardsAfterWaitTime` ever have an `error` passed in?

Reply:
So would it be better like this?
```
 combineShardsAfterWaitTime(waitTime, fileName, distinctShards) {
    return new Promise((resolve, reject) => {
      if (!fileName) reject(console.log("No file found"));
      setTimeout(() => resolve(fileUtils.assembleShards(fileName, distinctShards)), waitTime);
    });
  }
```

Reply:
Yeah, if an undefined filename would throw an error inside the promise code, then it seems like a good thing to hand. If the param `error` that will always be `undefined` then that wouldn't have been very useful I think. Not an expert, but a quick search seems to say that  it's [up to user](https://davidwalsh.name/promises) to reject promises, and they should have [promise rejection errors](https://eslint.org/docs/rules/prefer-promise-reject-errors).

Reply:
Updated with commit 3e1872c9043d5944ece48554fed829abedd7817e